Rakefile

```diff
@@ -66,16 +67,46 @@ directory PKGDIR
 directory "#{BUILDDIR}/parser_cache"
 file "#{BUILDDIR}/parser_cache" => BUILDDIR
 
-file "#{BUILDDIR}/foreman.yaml" => 'config/foreman.yaml' do |t|
-  cp t.prerequisites[0], t.name
-  sh 'sed -i "s#\(.*answer_file:\).*#\1 %s#" %s' % ["#{SYSCONFDIR}/foreman-installer/scenarios.d/foreman-answers.yaml", t.name]
-  sh 'sed -i "s#\(.*installer_dir:\).*#\1 %s#" %s' % ["#{DATADIR}/foreman-installer", t.name]
-  sh 'sed -i "s#\(.*log_dir:\).*#\1 %s#" %s' % ["#{LOGDIR}/foreman-installer", t.name]
-  sh 'sed -i "s#\(.*module_dirs:\).*#\1 %s#" %s' % ["#{DATADIR}/foreman-installer/modules", t.name]
-  sh 'sed -i "s#\(.*hiera_config:\).*#\1 %s#" %s' % ["#{DATADIR}/foreman-installer/config/foreman-hiera.conf", t.name]
-  sh 'sed -i "s#\(.*parser_cache_path:\).*#\1 %s#" %s' % ["#{DATADIR}/foreman-installer/parser_cache/foreman.yaml", t.name]
+SCENARIOS.each do |scenario|
+  scenario_config_replacements = {
+    'answer_file' => "#{SYSCONFDIR}/foreman-installer/scenarios.d/#{scenario}-answers.yaml",
+    'hiera_config' => "#{DATADIR}/foreman-installer/config/foreman-hiera.conf",
+    'installer_dir' => "#{DATADIR}/foreman-installer",
+    'log_dir' => "#{LOGDIR}/foreman-installer",
+    'module_dirs' => "#{DATADIR}/foreman-installer/modules",
+    'parser_cache_path' => "#{DATADIR}/foreman-installer/parser_cache/#{scenario}.yaml",
+  }
   if ENV['KAFO_MODULES_DIR']
-    sh 'sed -i "s#.*\(:kafo_modules_dir:\).*#\1 %s#" %s' % [ENV['KAFO_MODULES_DIR'], t.name]
+    scenario_config_replacements['kafo_modules_dir'] = ENV['KAFO_MODULES_DIR']
+  end
+
+  file "#{BUILDDIR}/#{scenario}.yaml" => "config/#{scenario}.yaml" do |t|
+    cp t.prerequisites[0], t.name
+    scenario_config_replacements.each do |setting, value|
+      sh 'sed -i "s#\(.*%s:\).*#\1 %s#" %s' % [setting, value, t.name]
```

Comment:
I think we actually use the scenario config to run tests as well. We can actually treat it as a YAML file because that's what the installer does at runtime anyway. That does mean we lose the comments but IMHO that's fine.

Reply:
I have verified this - we rely on correct values while building the parser cache. Templating then just means more work because you need to store an intermediate file.